engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -464,4 +474,55 @@ public static boolean removeAppender(String name)
     }
     return names;
   }
+
+  /**
+   * Makes MDC properties
+   */
+  public static void setupMDC()
+  {
+    String node = StramClientUtils.getHostName();
+    MDC.put("node", node == null ? "unknown" : node);
```

Comment:
The method java.net.getHostName() itself returns ip-address, if it is not able to retrieve the host name.

The implementation of our method getHostName() is:

  public static String getHostName()
  {
    try {
      return java.net.InetAddress.getLocalHost().getHostName();
    } catch (UnknownHostException ex) {
      return null;
    }
  }

The method java.net.InetAddress.getLocalHost() can throw the exception UnknownHostException if the local host could not be resolved into an address.